### ISSUE TITLE
Retroflag SNES Controller compatibility issues

### ISSUE THREAD

**Issue by remi-san** [godotengine/godot#43603]
**Godot version:** v3.2.3.stable.mono.official

**OS/device including version:** macOS Catalina 10.15.7

**Issue description:**

When connecting a retroflag SNES controller, buttons 13 and 15 (d-pad down and right) are like spammed and prevent buttons 12 and 14 (dpad up and left) to be detected when pressed (as a matter of fact, buttons 13 and 15 cannot be used either).

Gamepad GUID is **03000000830500006020000000010000**
Gamepad name is **iBuffalo SNES Controller**
The [SDL2 configuration](https://github.com/godotengine/godot/blob/fddbfdba91281d0368d451f64554b82cce347088/core/input/gamecontrollerdb.txt#L385) is already right, replacing it with `Input.AddJoyMapping(".....", true)` doesn't change that.

The controllers (as I've got two of them) have been tested with `gamepad-tool` which shows no spamming on the buttons and detects correctly all button presses. They also work well on other games on the same computer.

**Steps to reproduce:**
Connect a Retroflag SNES controller, detect in code (see snippet) which buttons have been pressed  and launch the scene in Godot.

```c#
        for (int i=0; i < 16; i++)
        {
            if (Input.IsJoyButtonPressed(0, i))
            {
                GD.Print("#" + i);
            }
        }

        GD.Print("------------");
```

It will print :
```
#13
#15
------------
#13
#15
------------
#13
#15
------------
```
endlessly, without any button pressed on the controller.

**Minimal reproduction project:**

[controller-problem.zip](https://github.com/godotengine/godot/files/5550030/controller-problem.zip)

**Comment by Calinou** [comment]
cc @madmiraal

**Comment by madmiraal** [comment]
Sounds like a duplicate of #41065. The specified controller mapping has D-pad buttons mapped to half-axes:
https://github.com/godotengine/godot/blob/bfed10185a55004b258eaad7fbb3e6049448edf9/core/input/gamecontrollerdb.txt#L385

#41065 was fixed with #42799, and backported to 3.2 (#42800) which will be included in 3.2.4.beta2.
